bodhi/server/util.py

```diff
@@ -684,3 +686,101 @@ def sort_severity(value):
+
+        for branch in pdc_request_json['results']:
+            critpath_pkgs_set.add(branch['global_component'])
+
+        if pdc_request_json['next']:
+            pdc_api_url_with_args = pdc_request_json['next']
+        else:
+            # There are no more results to iterate through
+            break
+    return list(critpath_pkgs_set)
+
+
+def pagure_api_get(pagure_api_url, timeout=60):
+    """
+    A wrapper to get API resources from Pagure
+    :param pagure_api_url: a string of the URL to query
+    :param timeout: the timeout on the GET request
+    :return: JSON of the API resource(s) or a RuntimeError
+    """
+    base_error_msg = (
+        'Bodhi failed to get a resource from Pagure at the following URL '
+        '"{0}". The status code was "{1}"')
+    rv = requests.get(pagure_api_url, timeout=timeout)
+
+    if rv.status_code == 200:
+        return rv.json()
+    elif rv.status_code == 500:
+        # There will be no JSON with an error message here
+        error_msg = base_error_msg.format(pagure_api_url, rv.status_code)
+        log.exception(error_msg)
+        raise RuntimeError(
+            base_error_msg.format(pagure_api_url, rv.status_code))
+    else:
+        # If it's not a 500 error, we can assume that Pagure returned an error
+        # message in JSON that we can log
+        try:
+            rv_error = rv.json().get('error')
+        except ValueError:
+            rv_error = ''
+        error_msg = base_error_msg.format(pagure_api_url, rv.status_code)
+        error_msg = '{0}. The error was "{1}"'.format(error_msg, rv_error)
+        log.exception(error_msg)
+        raise RuntimeError(error_msg)
+
+
+def pdc_api_get(pdc_api_url, timeout=60):
+    """
+    A wrapper to get API resources from PDC
+    :param pdc_api_url: a string of the URL to query
+    :param timeout: the timeout on the GET request
+    :return: JSON of the API resource(s) or a RuntimeError
+    """
+    base_error_msg = (
+        'Bodhi failed to get a resource from PDC at the following URL '
+        '"{0}". The status code was "{1}"')
+    rv = requests.get(pdc_api_url, timeout=timeout)
+
+    if rv.status_code == 200:
+        return rv.json()
+    elif rv.status_code == 500:
+        # There will be no JSON with an error message here
+        error_msg = base_error_msg.format(pdc_api_url, rv.status_code)
+        log.exception(error_msg)
+        raise RuntimeError(
+            base_error_msg.format(pdc_api_url, rv.status_code))
+    else:
+        # If it's not a 500 error, we can assume that PDC returned an error
+        # message in JSON that we can log
+        try:
+            rv_error = rv.json()
+        except ValueError:
+            rv_error = ''
+        error_msg = base_error_msg.format(pdc_api_url, rv.status_code)
+        error_msg = '{0}. The error was "{1}"'.format(error_msg, rv_error)
+        log.exception(error_msg)
+        raise RuntimeError(error_msg)
```

Comment:
This function is extremely similar to the one for Pagure handling - the only difference I notice is whether we know the error from the json or not. Because of this, I recommend making a single unified helper function they can both use, and have each of these functions just use that. Then the special error handling can happen at the higher layer and we don't have so much copied code.